x-pack/platform/plugins/shared/embeddable_alerts_table/public/plugin.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CoreSetup, CoreStart, Plugin } from '@kbn/core/public';
+import { EMBEDDABLE_ALERTS_TABLE_ID } from './constants';
+import type {
+  EmbeddableAlertsTablePluginSetup,
+  EmbeddableAlertsTablePluginStart,
+  SetupDependencies,
+  StartDependencies,
+} from './types';
+
+export class EmbeddableAlertsTablePlugin
+  implements Plugin<EmbeddableAlertsTablePluginSetup, EmbeddableAlertsTablePluginStart>
+{
+  public setup(core: CoreSetup<StartDependencies>, { embeddable }: SetupDependencies) {
+    embeddable.registerReactEmbeddableFactory(EMBEDDABLE_ALERTS_TABLE_ID, async () => {
```

Comment:
Let's move the registration to a separate file and call a function like `registerAlertsTableEmbeddableFactory({ embeddable, core })`.